src/app/clusters/actions-server/actions-server.cpp

```diff
@@ -0,0 +1,462 @@
+        return;
+    case Actions::Commands::PauseAction::Id:
+        HandleCommand<Commands::PauseAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandlePauseAction(ctx, commandData); });
+        return;
+    case Actions::Commands::PauseActionWithDuration::Id:
+        HandleCommand<Commands::PauseActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandlePauseActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::ResumeAction::Id:
+        HandleCommand<Commands::ResumeAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleResumeAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableAction::Id:
+        HandleCommand<Commands::EnableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleEnableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableActionWithDuration::Id:
+        HandleCommand<Commands::EnableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleEnableActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableAction::Id:
+        HandleCommand<Commands::DisableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleDisableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableActionWithDuration::Id:
+        HandleCommand<Commands::DisableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDisableActionWithDuration(ctx, commandData); });
+        return;
+    }
+}
+
+CHIP_ERROR ActionsServer::ModifyActionList(EndpointId aEndpoint, const ActionStructStorage & aAction)
+{
+    Delegate * delegate = GetDelegate(aEndpoint);
+    VerifyOrReturnError(IsValid(delegate) == CHIP_NO_ERROR, CHIP_ERROR_INCORRECT_STATE);
+
+    // Read through the list to find and update the existing action that matches the passed-in action's ID.
+    for (uint16_t i = 0; i < kMaxActionListLength; i++)
+    {
+        ActionStructStorage existingAction;
+        CHIP_ERROR err = delegate->ReadActionAtIndex(i, existingAction);
+        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+        {
+            break;
+        }
+
+        if (existingAction.actionID == aAction.actionID)
+        {
+            existingAction.Set(aAction.actionID, aAction.name, aAction.type, aAction.endpointListID, aAction.supportedCommands,
+                               aAction.state);
+
+            MarkDirty(aEndpoint, Attributes::ActionList::Id);
```

Comment:
I don't understand this code.  The Set() is changing an on-stack data structure that is never used for anything after that `Set()` call.  Why is that `Set()` call there and what is this trying to do?

Reply:
Yes, as we discussed earlier, only the `application` can change/update the `ActionList` and `EndpointLists` attributes. So whenever there is a change respective `Modify*` APIs should be called that will update the `on-stack data and mark an attribute dirty`.

Reply:
Why are we updating "on-stack data"?  No one will ever see that data.

If this API is meant to be a "notify the cluster that the backing storage for this attribute has changed" API, then it should be named appropriately, and does not need the "aAction" passed in at all, does it?

Reply:
I have changed the API to a notification and removed `aAction` from the parameter.